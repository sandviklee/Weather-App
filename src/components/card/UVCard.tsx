import styles from "./UVCard.module.css";

interface CardProps {
  UVindex: number;
}

const UVCard = ({ UVindex }: CardProps) => {
  return (
    <div className={styles.container}>
      <p className={styles.title}>UV Index</p>
      <div className={styles.bar_container}>
        <div className={styles.bar}>
          <div
            style={{
              backgroundColor: "#FF8E3C",
              height: "100%",
              width: `${(UVindex / 6) * 100}%`,
            }}
          ></div>
        </div>
        <p className={styles.uv_index}>{UVindex}</p>
      </div>
    </div>
  );
};

export default UVCard;
