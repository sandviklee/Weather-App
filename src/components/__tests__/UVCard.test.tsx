import { render, screen } from '@testing-library/react';
import UVCard from '../card/UVCard';

describe('UVCard component', () => {
    test('renders correctly', () => {
        const testUVIndex = 3;
        render(<UVCard UVindex={testUVIndex} />);
        
        const uvIndexTitle = screen.getByText('UV Index');
        expect(uvIndexTitle).toBeInTheDocument();
        const uvIndexValue = screen.getByText(testUVIndex.toString());
        expect(uvIndexValue).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<UVCard UVindex={3} />);
        expect(asFragment()).toMatchSnapshot();
    });
});