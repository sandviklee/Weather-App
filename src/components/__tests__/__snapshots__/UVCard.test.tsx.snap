// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UVCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_a120ba"
  >
    <p
      class="_title_a120ba"
    >
      UV Index
    </p>
    <div
      class="_bar_container_a120ba"
    >
      <div
        class="_bar_a120ba"
      >
        <div
          style="background-color: rgb(255, 142, 60); height: 100%; width: 50%;"
        />
      </div>
      <p
        class="_uv_index_a120ba"
      >
        3
      </p>
    </div>
  </div>
</DocumentFragment>
`;
